src/ws/lobby.js

```diff
@@ -11,66 +11,74 @@ function find(connections, sessions, ws, participantId, sessionId) {
         }
     }
 
-    if (participantId && sessionId) {
-        const session = sessions.get(sessionId);
-
-        if (session) {
-            const connection = session.get(participantId);
-
-            if (connection) {
-                return [connection, participantId, sessionId];
+    if (participantId && sessionAlias && game) {
+        const gameSession = gameSessions.get(game);
+        if (gameSession) {
+            const session = gameSession.get(sessionAlias);
+            if (session) {
+                const connection = session.get(participantId);
+                if (connection) {
```

Comment:
maybe it makes sense to use early return? 4 levels if nesting is kinda too much defensive 😄

Reply:
Where is `Maybe` when you really need it?..

Reply:
I believe soon with `a?.b?.c`

Reply:
Elvis operator, just like CoffeeScript ☕️  
That's shorter syntax but still kinda meh IMHO

Reply:
```hs
do
    gameSession <- get gameSessions game
    session <- get gameSession sessionAlias
    connection <- get session participantId
    return [connection, participantId, sessionAlias, game]
```

Reply:
No need to do anything if the chain is broken on the 1st step ⚡️

Reply:
Yeah, exactly. In the example above if any of the `get`s returns `Nothing` the whole `do` returns `Nothing`

Reply:
It doesn't work :)

![image](https://user-images.githubusercontent.com/3236551/30867540-8dc9f342-a2dc-11e7-887a-04bd7c8ad3f3.png)

Reply:
![](http://weknowmemes.com/wp-content/uploads/2012/03/soon-meme.jpg)

Reply:
Option 1 - 👍, Option 2 - ❤️

Option 1
```
    if (!participantId || !sessionAlias || !game) {
        return null;
    }

    const gameSession = gameSessions.get(game);
    if (!gameSession) {
        return null;
    }

    const session = gameSession.get(sessionAlias);
    if (!session) {
        return null;
    }

    const connection = session.get(participantId);
    if (!connection) {
        return null;
    }

    return [connection, participantId, sessionAlias, game];
```

Option 2
```
    if (participantId && sessionAlias && game) {
        const gameSession = gameSessions.get(game);
        if (gameSession) {
            const session = gameSession.get(sessionAlias);
            if (session) {
                const connection = session.get(participantId);
                if (connection) {
                    return [connection, participantId, sessionAlias, game];
                }
            }
        }
    }

    return null;
```